packages/azure-kusto-ingest/src/ingestClient.browser.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { KustoConnectionStringBuilder } from "azure-kusto-data";
+
+import { BlobDescriptor, StreamDescriptor } from "./descriptors";
+import { FileDescriptor } from "./fileDescriptor";
+import { QueueSendMessageResponse } from "@azure/storage-queue";
+
+import { IngestionPropertiesInput } from "./ingestionProperties";
+import { KustoIngestClientBase } from "./ingestClientBase";
+import { BlockBlobUploadOptions } from "@azure/storage-blob";
+
+export class KustoIngestClient extends KustoIngestClientBase {
+    constructor(kcsb: string | KustoConnectionStringBuilder, defaultProps?: IngestionPropertiesInput) {
+        super(kcsb, defaultProps, true);
+    }
+
+    /**
+     * Use string for Node.js and Blob for browser
+     */
+    async ingestFromFile(file: FileDescriptor | Blob | string, ingestionProperties?: IngestionPropertiesInput): Promise<QueueSendMessageResponse> {
+        this.ensureOpen();
+        if (!(file instanceof Blob) && !((file as FileDescriptor).file instanceof Blob)) {
```

Comment:
You can't protect this at the type level?

Reply:
The abstraction limits my ability to change the types here

Reply:
If base accepts "any", and we widen the accepted types in the inherited, wouldn't it just work? won't break either..